### PR TITLE
ECO-3120: Added notification table, added/updated all necessary models.

### PR DESCRIPTION
**Pull request by aleksey-kotsuba**
CR - https://spryker.atlassian.net/browse/ECO-3069

### CODE REVIEW

File: config/Zed/Oms/HeidelpayInvoiceSecuredB2c01.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="spryker:oms-01 http://static.spryker.com/oms-01.xsd">
 
-    <process name="HeidelpayInvoiceSecuredB2C01" main="true">
+    <process name="HeidelpayInvoiceSecuredB2c01" main="true">
```
Comment (gechetspr) [inline config/Zed/Oms/HeidelpayInvoiceSecuredB2c01.xml]: a little bit strange change

File: src/SprykerEco/Shared/Heidelpay/HeidelpayConstants.php [hunk 2]
```diff
@@ -28,6 +28,7 @@ interface HeidelpayConstants
     public const CONFIG_HEIDELPAY_TRANSACTION_CHANNEL_IDEAL = 'HEIDELPAY:CONFIG_HEIDELPAY_TRANSACTION_CHANNEL_IDEAL';
     public const CONFIG_HEIDELPAY_TRANSACTION_CHANNEL_SOFORT = 'HEIDELPAY:CONFIG_HEIDELPAY_TRANSACTION_CHANNEL_SOFORT';
     public const CONFIG_HEIDELPAY_TRANSACTION_CHANNEL_EASY_CREDIT = 'HEIDELPAY:CONFIG_HEIDELPAY_TRANSACTION_CHANNEL_EASY_CREDIT';
+    public const CONFIG_HEIDELPAY_TRANSACTION_CHANNEL_INVOICE_SECURED_B2C = 'HEIDELPAY:CONFIG_HEIDELPAY_TRANSACTION_CHANNEL_INVOICE_SECURED_B2C';
```
Comment (gechetspr) [inline src/SprykerEco/Shared/Heidelpay/HeidelpayConstants.php]: Please add specification and api tag

File: src/SprykerEco/Shared/Heidelpay/Transfer/heidelpay.transfer.xml [hunk 3]
```diff
@@ -232,7 +232,7 @@
         <property name="heidelpayPaypalAuthorize" type="HeidelpayPayment"/>
         <property name="heidelpayPaypalDebit" type="HeidelpayPayment"/>
         <property name="heidelpayCreditCardSecure" type="HeidelpayCreditCardPayment"/>
-        <property name="heidelpayInvoiceSecuredB2C" type="HeidelpayPayment"/>
+        <property name="heidelpayInvoiceSecuredB2c" type="HeidelpayPayment"/>
```
Comment (gechetspr) [inline src/SprykerEco/Shared/Heidelpay/Transfer/heidelpay.transfer.xml]: same

File: src/SprykerEco/Shared/Heidelpay/HeidelpayConfig.php [hunk 4]
```diff
@@ -19,7 +19,7 @@ class HeidelpayConfig extends AbstractBundleConfig
     public const PAYMENT_METHOD_PAYPAL_DEBIT = self::PROVIDER_NAME . 'PaypalDebit';
     public const PAYMENT_METHOD_IDEAL = self::PROVIDER_NAME . 'Ideal';
     public const PAYMENT_METHOD_EASY_CREDIT = self::PROVIDER_NAME . 'EasyCredit';
-    public const PAYMENT_METHOD_INVOICE_SECURED_B2C = self::PROVIDER_NAME . 'InvoiceSecuredB2C';
+    public const PAYMENT_METHOD_INVOICE_SECURED_B2C = self::PROVIDER_NAME . 'InvoiceSecuredB2c';
```
Comment (gechetspr) [inline src/SprykerEco/Shared/Heidelpay/HeidelpayConfig.php]: same

File: src/SprykerEco/Yves/Heidelpay/Form/DataProvider/InvoiceSecuredB2cDataProvider.php [hunk 5]
```diff
@@ -26,11 +26,11 @@ public function getData(AbstractTransfer $quoteTransfer): QuoteTransfer
             $quoteTransfer->setPayment(new PaymentTransfer());
         }
 
-        if ($quoteTransfer->getPayment()->getHeidelpayInvoiceSecuredB2C() !== null) {
+        if ($quoteTransfer->getPayment()->getHeidelpayInvoiceSecuredB2c() !== null) {
```
Comment (gechetspr) [inline src/SprykerEco/Yves/Heidelpay/Form/DataProvider/InvoiceSecuredB2cDataProvider.php]: Why it was done in the first place? it changes a lot, but there is no real value in this change

File: src/SprykerEco/Zed/Heidelpay/Persistence/Mapper/HeidelpayPersistenceMapper.php [hunk 6]
```diff
@@ -0,0 +1,32 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerEco\Zed\Heidelpay\Persistence\Mapper;
```
Comment (gechetspr) [inline src/SprykerEco/Zed/Heidelpay/Persistence/Mapper/HeidelpayPersistenceMapper.php]: please put it in the propel folder

File: src/SprykerEco/Zed/Heidelpay/Persistence/Propel/Schema/spy_heidelpay.schema.xml [hunk 7]
```diff
@@ -93,4 +93,37 @@
         </index>
     </table>
 
+    <table name="spy_payment_heidelpay_notification">
```
Comment (gechetspr) [inline src/SprykerEco/Zed/Heidelpay/Persistence/Propel/Schema/spy_heidelpay.schema.xml]: as long as it a new table, please name it according to our conventions

File: src/SprykerEco/Zed/Heidelpay/Persistence/Propel/Schema/spy_heidelpay.schema.xml [hunk 8]
```diff
@@ -93,4 +93,37 @@
...
+
+        <column name="result_code" required="true" type="VARCHAR" />
+        <column name="result_timestamp" required="false" type="VARCHAR" />
+        <column name="result" required="true" type="VARCHAR" />
+        <column name="result_status" required="false" type="VARCHAR" />
+        <column name="result_reason" required="false" type="VARCHAR" />
+        <column name="result_return" required="false" type="VARCHAR" />
+
+        <column name="payment_code" required="false" type="VARCHAR" />
+        <column name="amount" required="false" type="VARCHAR" />
```
Comment (gechetspr) [inline src/SprykerEco/Zed/Heidelpay/Persistence/Propel/Schema/spy_heidelpay.schema.xml]: so even amount is varchar?

File: config/Zed/Oms/HeidelpayInvoiceSecuredB2c01.xml (hunk 1 above)
Comment (aleksey-kotsuba) [reply config/Zed/Oms/HeidelpayInvoiceSecuredB2c01.xml]: That was requested by @alex-galych.

File: src/SprykerEco/Zed/Heidelpay/Persistence/Propel/Schema/spy_heidelpay.schema.xml (hunk 7 above)
Comment (aleksey-kotsuba) [reply src/SprykerEco/Zed/Heidelpay/Persistence/Propel/Schema/spy_heidelpay.schema.xml]: I don't think that is a good ideal. In this scenario we won't have all heidelpay related tables in the same place.

File: src/SprykerEco/Zed/Heidelpay/Persistence/Propel/Schema/spy_heidelpay.schema.xml (hunk 8 above)
Comment (aleksey-kotsuba) [reply src/SprykerEco/Zed/Heidelpay/Persistence/Propel/Schema/spy_heidelpay.schema.xml]: We are receiving this in xml and it converts to string by default. We can cast it to float or int but IMO it's not necessary. We won't use it as a value for any calculations or something.

**Pull request merged**